### PR TITLE
make auto crop default be page content

### PR DESCRIPTION
**Pull request by hugleo**
Based on:

```
function ReaderCropping:setCropZoomMode(confirmed)
    if confirmed then
        -- if original zoom mode is "page???", set zoom mode to "content???"
        local zoom_mode_type = self.orig_zoom_mode:match("page(.*)")
        self:setZoomMode(zoom_mode_type
                    and "content"..zoom_mode_type
                    or self.orig_zoom_mode)
        self.ui:handleEvent(Event:new("InitScrollPageStates"))
    else
        self:setZoomMode(self.orig_zoom_mode)
    end
end
```

- - -
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader/11438)

### CODE REVIEW

**Comment by Frenzie** [comment]
What concretely does this mean in terms of old and new behavior?

**Comment by hugleo** [comment]
> What concretely does this mean in terms of old and new behavior?

Auto crop is the default. When we first the open the document (with empty settings, no sdr) the document is not cropped because is in page mode. We have to click in Page crop auto to apply the settings.

**Comment by Frenzie** [comment]
So the old behavior is that the second time it would do autocrop?

**Comment by hugleo** [comment]
> So the old behavior is that the second time it would do autocrop?

Not even that. Will crop only if we click in the autocrop button that's when what it's needed is configured here:
https://github.com/koreader/koreader/blob/master/frontend/apps/reader/modules/readercropping.lua#L158

**Comment by Frenzie** [comment]
I'm fine with the change, but it sounds more like a change in behavior than correctly doing what the code's going for in that case. :-)

**Comment by hugleo** [comment]
> I'm fine with the change, but it sounds more like a change in behavior than correctly doing what the code's going for in that case. :-)

Once we click on page crop button (["zoom_mode"] = "content" will be forced) we can't go back to the default behavior anymore. We have to reset the documents settings or edit the sdr parts like zoom_mode = "pagewidth".

If auto-crop requires content mode (https://github.com/koreader/koreader/blob/master/frontend/apps/reader/modules/readercropping.lua#L156) then enforce on the "first open" as well.

**Comment by NiLuJe** [comment]
Yup, noticed that inconsistency every time I delved into ReaderZoom hell.

File: frontend/apps/reader/modules/readerzooming.lua [hunk 1]
```diff
@@ -240,6 +240,12 @@ function ReaderZooming:onReadSettings(config)
         zoom_mode = self.zoom_mode_label[zoom_mode] and zoom_mode or self.DEFAULT_ZOOM_MODE
     end
 
+    -- auto crop needs to be page content
+    local trim_page = config:readSetting("kopt_trim_page")
+                             or G_reader_settings:readSetting("kopt_trim_page") or G_defaults:readSetting("DKOPTREADER_CONFIG_TRIM_PAGE")
```
Comment (poire-z) [inline frontend/apps/reader/modules/readerzooming.lua:245]: Please lake the last `or ...` on another aligned on the previous `or` :)

**Comment by poire-z** [comment]
^ Some Circle CI issues that need fixing (either in the code if they are relevant, or in the tests themselves if they should test different or expect different results).

**Comment by Frenzie** [comment]
In context presumably the tests.

**Comment by hugleo** [comment]
This thing has multiple possibilities, this is one of them :)

**Review by poire-z — APPROVED** [review]
Reads simple(r) and fine :)

**Comment by poire-z** [comment]
Good to merge? No after thought after 3 weeks ? :)
You also have https://github.com/koreader/libk2pdfopt/pull/44 that has not yet been bumped into koreader-base. Can you do it or should we do it for you?

**Comment by hugleo** [comment]
> Good to merge? No after thought after 3 weeks ? :) 

I think so. Didn't thought about this these weeks :)

> You also have [koreader/libk2pdfopt#44](https://github.com/koreader/libk2pdfopt/pull/44) that has not yet been bumped into koreader-base. Can you do it or should we do it for you?

Need to merge another approved PR in libk2pdfopt before  that.

**Comment by poire-z** [comment]
koreader-base bump up to you.

**Pull request merged**